ocaml/parsing/parser.mly

```diff
@@ -3282,9 +3284,14 @@ pattern_comma_list(self):
         Some label, mkpat_opt_constraint ~loc pat (Some cty) }
 ;
 
-labeled_pattern_comma_list(self):
-    labeled_pattern_comma_list(self) SEMI labeled_pattern      { $3 :: $1 }
-  | labeled_pattern SEMI labeled_pattern                       { [$3; $1] }
+labeled_pattern_comma_list:
+    DOTDOT
+      { [], Open }
+  | labeled_pattern SEMI labeled_pattern
+      { [$1; $3], Closed }
+  | labeled_pattern SEMI labeled_pattern_comma_list
+      { let l, closed = $3 in
+        $1 :: l, closed }
```

Comment:
Sounds good.  Can you add tests demonstrating these are rejected by the parser?

It can be tricky to do parse errors using the expect test framework - you might need to do this using the other mechanism, where the expected output is a reference file instead.  See `tests/parse-errors` for some examples.